### ISSUE TITLE
[HOLD for payment 2023-07-24] [$1000] The behavior of the delete message function in the workspace's announce room is not consistent.

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#21508]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. User and Admin are in the workspace's announce room
2. User send the message A
3. Admin (or anyone) replies to message A
4. User send OTHER message B (not a reply)
5. Admin delete ALL Messages and Replies
## Expected Result: 
All message should be disappear or at least, the behaviour should be the same between deleting messages and deleting replies
## Actual Result: 
The label [Deleted message] is still displaying if we delete the Message and all of its Replies. BUT the not displaying for the normal message (which HAS NO replies)

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.32-5
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/9dd84215-07ee-442c-846d-b86555b4d6ed

**Expensify/Expensify Issue URL:**
**Issue reported by:** @tranvantoan-qn
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687262155164809

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01637d9a166a023628</li>
        <li>Upwork Job ID: 1673432730642673664</li>
        <li>Last Price Increase: 2023-06-26</li>
    </ul>
</details>

**Comment by tranvantoan-qn** [comment]
https://github.com/Expensify/App/assets/10254571/7a53e570-aab8-41d3-bd93-4d22ea7eb159

**Comment by KALLL** [comment]
Kindly it is a dupe of https://github.com/Expensify/App/issues/21103

**Comment by tranvantoan-qn** [comment]
@KALLL 
It's **different**. 
- My case is about deleting all messages - including the ones inside the thread  - and it always happens
- In my case, **there is NO delete option** if a message is deleted (see attached image)- So there is no way for me to reproduce the same case as you

![image](https://github.com/Expensify/App/assets/10254571/deecbf37-3270-4a87-bddb-c001f548e58b)

So let @kavimuru and @Christinadobrzyn review to see **if the solution applied for your issue** will resolve both then decide it this is DUPE.?

**Comment by Christinadobrzyn** [comment]
I think this is different from https://github.com/Expensify/App/issues/21103 but I wonder if it's also a regression from https://github.com/Expensify/App/pull/20460

Adding eng and External to have more thoughts.

**Comment by dangrous** [comment]
So yeah I think this is different from #21103, and I'm pretty sure the solution to that won't fix this.

I (personally) think that we should always show the `[Deleted message]` and should never fully remove a message, but we've decided the other way in multiple conversations so I'll be quiet on this front 😄 

That being said, I do agree that in this case, it makes sense to fully delete the parent of a thread if the thread no longer exists. I think, though, this would be a backend fix, as - for example, you actually don't even have to ever have a thread. If you:
1. Write a message
2. Click the "Reply in thread" button
3. Delete the original message
You still get the same behavior.

@mananjadhav let me know if you agree - or if we should open it up externally. If it's backend, I can look into a fix

**Comment by Piotrfj** [comment]
I'm not sure about the status of this bug. Is it open for external?

**Comment by dangrous** [comment]
> I'm not sure about the status of this bug. Is it open for external?

@mananjadhav when you have a sec can you check [my work](https://github.com/Expensify/App/issues/21508#issuecomment-1610037358)? I'm pretty sure this is internal, but want to double confirm before we remove `Help Wanted`

**Comment by mananjadhav** [comment]
@dangrous I think this would be part backend and part App? The offline mode, and optimistic calls would have to be done on  the App side? while the thread deletion would've have to be done on the backend?

**Comment by dangrous** [comment]
Okay cool, that makes sense. I'm going to switch this to Internal and remove Help Wanted, and then we can bring that back once the backend is complete!

Now to actually start figuring out what I need to do haha

**Comment by dangrous** [comment]
Haven't had a moment to go into this yet, should hopefully be able to today, if not, then definitely this week

**Comment by dangrous** [comment]
Tomorrow or later today for sure!

**Comment by dangrous** [comment]
Blah haven't had the time I wanted

**Comment by dangrous** [comment]
I have a fix in progress, am testing it! Might need an assist from some people who know a bit more about threads but can handle it I think!

**Comment by Christinadobrzyn** [comment]
Should I move this to a weekly label?

**Comment by dangrous** [comment]
eh, I should be able to get the code in review today, so I think we'll be fine either way

**Comment by dangrous** [comment]
Okay I went through a whole thing. Turns out this isn't actually internal, and can be fixed just on App. But I won't make this external again since I figured it out (see PR!).

I'm pretty sure this does the trick and doesn't make anything weird - unless I'm totally off base this was happening on all chats, not just announce rooms. So I fixed it for all.

**Comment by Christinadobrzyn** [comment]
I'm going to be ooo until August 31st so going to unassign and assign a new teammate. 

At this time, we're receiving and reviewing proposals.

**Comment by mananjadhav** [comment]
I couldn't pinpoint the point at which this would've broken? May be we never intended to add this? @dangrous are you aware? 

Regardless I think it makes sense to add a regression test for this one because we've had discussions on whether to show `[Deleted Message]`, and if in case we want to change the behavior, this shouldn't be reported again.

The QA steps from the PR are good enough and detailed. @dangrous  @jliexpensify @Christinadobrzyn wdyt?

@jliexpensify Because this is internal, only reporting bonus to be paid out and C+ review payout I'll raise on NewDot.

**Comment by jliexpensify** [comment]
@tranvantoan-qn - can you apply here? https://www.upwork.com/jobs/~01637d9a166a023628

I'm having issues inviting you - thanks!

**Comment by jliexpensify** [comment]
@mananjadhav I think @dangrous is probably best equipped to answer your questions here!

**Comment by tranvantoan-qn** [comment]
@jliexpensify 
I've applied for the job, Thank you!

**Comment by jliexpensify** [comment]
Paid @tranvantoan-qn , just awaiting Daniel to answer Manan :)

**Comment by mananjadhav** [comment]
I have raised a request for the payout on NewDot. @dangrous @jliexpensify

**Comment by jliexpensify** [comment]
@mananjadhav @dangrous so this is what i should add to the QA GH?

![image](https://github.com/Expensify/App/assets/30815269/b742090b-4dba-4a8b-8908-7da661e26d91)

**Comment by dangrous** [comment]
You can drop step 8, which I think was a clarification rather than an actual step. Otherwise yep!

**Comment by jliexpensify** [comment]
Done!

**Comment by JmillsExpensify** [comment]
@jliexpensify  Can you please summarize the appropriate individual payments for all parties involved in this issue? This is holding up @mananjadhav's NewDot payments. More information on this compliance process [in Slack](https://expensify.slack.com/archives/C01SKUP7QR0/p1690321157273869).

**Comment by jliexpensify** [comment]
Payments:

- Issue Reporter: @tranvantoan-qn $250
- C+: @mananjadhav $1000

Upworks job is [here](https://www.upwork.com/jobs/~01637d9a166a023628).

**Comment by JmillsExpensify** [comment]
Reviewed details for @mananjadhav. This is accurate based on summary from Business Reviewer and approved for payment in NewDot.
